firmware/Core/Src/telecommands/telecommand_adcs.c

```diff
@@ -0,0 +1,1646 @@
+        args_8[i] = (uint8_t) arguments[i];
+    }
+    
+    uint8_t status = ADCS_Set_Power_Control((ADCS_Power_Select) args_8[0], (ADCS_Power_Select) args_8[1], (ADCS_Power_Select) args_8[2], (ADCS_Power_Select) args_8[3], (ADCS_Power_Select) args_8[4], (ADCS_Power_Select) args_8[5], (ADCS_Power_Select) args_8[6], (ADCS_Power_Select) args_8[7], (ADCS_Power_Select) args_8[8], (ADCS_Power_Select) args_8[9]);
+    return status;
+}                            
+
+/// @brief Telecommand: Request the given telemetry data from the ADCS
+/// @param args_str 
+///     - Arg 0: Mounting transform alpha angle [deg] 
+///     - Arg 1: Mounting transform beta angle [deg]
+///     - Arg 2: Mounting transform gamma angle [deg]
+///     - Arg 3: Channel 1 offset value
+///     - Arg 4: Channel 2 offset value
+///     - Arg 5: Channel 3 offset value
+///     - Arg 6: Value (1, 1) of the magnetometer sensitivity matrix
+///     - Arg 7: Value (2, 2) of the magnetometer sensitivity matrix
+///     - Arg 8: Value (3, 3) of the magnetometer sensitivity matrix
+///     - Arg 9: Value (1, 2) of the magnetometer sensitivity matrix
+///     - Arg 10: Value (1, 3) of the magnetometer sensitivity matrix
+///     - Arg 11: Value (2, 1) of the magnetometer sensitivity matrix
+///     - Arg 12: Value (2, 3) of the magnetometer sensitivity matrix
+///     - Arg 13: Value (3, 1) of the magnetometer sensitivity matrix
+///     - Arg 14: Value (3, 2) of the magnetometer sensitivity matrix
+/// @return 0 on success, >0 on error
+uint8_t TCMDEXEC_adcs_set_magnetometer_config(const char *args_str, TCMD_TelecommandChannel_enum_t tcmd_channel,
+                                              char *response_output_buf, uint16_t response_output_buf_len) {
+
+    // parse arguments into doubles
+    const uint8_t num_args = 15;
+    double arguments[num_args]; 
+    for (uint8_t i = 0; i < num_args; i++) {
+        TCMD_extract_double_arg(args_str, strlen(args_str), i, &arguments[i]);
+    }
+    
+    uint8_t status = ADCS_Set_Magnetometer_Config(arguments[0], arguments[1], arguments[2], arguments[3], arguments[4], arguments[5], arguments[6], arguments[7], arguments[8], arguments[9], arguments[10], arguments[11], arguments[12], arguments[13], arguments[14]);
+    return status;
+}     
+
+/// @brief Telecommand: Request the given telemetry data from the ADCS
+/// @param args_str 
+///     - No arguments for this command
+/// @return 0 on success, >0 on error
+uint8_t TCMDEXEC_adcs_bootloader_clear_errors(const char *args_str, TCMD_TelecommandChannel_enum_t tcmd_channel,
+                                              char *response_output_buf, uint16_t response_output_buf_len) {
+    uint8_t status = ADCS_Bootloader_Clear_Errors(); 
+    return status;
+}                                    
+
+/// @brief Telecommand: Request the given telemetry data from the ADCS
+/// @param args_str 
+///     - Arg 0: whether to save the current Unix time immediately
```

Comment:
Explain that these are booleans passed as ints (0/1). Perhaps `(1 = save immediately, 0 = don't save at all)` ?

Reply:
This is done.